src/common/admin_socket_client.cc

```diff
@@ -169,6 +171,7 @@ std::string AdminSocketClient::do_request(std::string request, std::string *resu
     err = oss.str();
     goto done;
   }
+  vec[message_size] = '\0';
```

Comment:
instead of using `result->assign((const char*)buffer);`, probably we should use 
```c++
result->assign(buffer, message_size);
```

or probably better, just define the `vec` as a `string`, so we can:
```c++
string buffer;
...
buffer.reserve(message_size);
res = safe_read_exact(socket_fd, &buffer[0], message_size);
...
swap(*result, buffer);
```
so we can remove `buffer`, and have less memcpy() calls.